src/jit/SimdArm64Inl.h

```diff
@@ -893,3 +910,165 @@ static void emitShuffleSIMD(sljit_compiler* compiler, Instruction* instr)
         sljit_emit_simd_mov(compiler, SLJIT_SIMD_STORE | SLJIT_SIMD_REG_128 | SLJIT_SIMD_ELEM_128, dst, args[2].arg, args[2].argw);
     }
 }
+
+static void emitShiftSIMD(sljit_compiler* compiler, Instruction* instr)
+{
+    Operand* operands = instr->operands();
+    JITArg args[3];
+
+    sljit_s32 type = SLJIT_SIMD_ELEM_128;
+
+    int mask = 0;
+
+    switch (instr->opcode()) {
+    case ByteCode::I8X16ShlOpcode:
+    case ByteCode::I8X16ShrSOpcode:
+    case ByteCode::I8X16ShrUOpcode: {
+        mask = SimdOp::shiftB;
+        type = SLJIT_SIMD_ELEM_8;
+        break;
+    }
+    case ByteCode::I16X8ShlOpcode:
+    case ByteCode::I16X8ShrSOpcode:
+    case ByteCode::I16X8ShrUOpcode: {
+        mask = SimdOp::shiftH;
+        type = SLJIT_SIMD_ELEM_16;
+        break;
+    }
+    case ByteCode::I32X4ShlOpcode:
+    case ByteCode::I32X4ShrSOpcode:
+    case ByteCode::I32X4ShrUOpcode: {
+        mask = SimdOp::shiftS;
+        type = SLJIT_SIMD_ELEM_32;
+        break;
+    }
+    case ByteCode::I64X2ShlOpcode:
+    case ByteCode::I64X2ShrSOpcode:
+    case ByteCode::I64X2ShrUOpcode: {
+        mask = SimdOp::shiftD;
+        type = SLJIT_SIMD_ELEM_64;
+        break;
+    }
+    default:
+        ASSERT_NOT_REACHED();
+    }
+
+    simdOperandToArg(compiler, operands, args[0], type, SLJIT_FR0);
+    args[1].set(operands + 1);
+
+    args[2].set(operands + 2);
+    sljit_s32 dst = GET_TARGET_REG(args[2].arg, SLJIT_FR1);
+
+    bool isImm = false;
+
+    if (((args[1].arg) & (SLJIT_IMM))) {
```

Comment:
There is a macro for this to check in the compiler. I don't think we need an extra boolean here.